lib/socket.c

```diff
@@ -28,6 +28,16 @@
 #define HAVE_SYS_SOCKET_H 1
 #endif
 
+#if !defined(__amigaos4__) && (defined(__AMIGA__) || defined(__AROS__))
+#include <sys/ioctl.h>
+#include <proto/bsdsocket.h>
+#undef getaddrinfo
+#undef freeaddrinfo
+#undef HAVE_UNISTD_H
+#define close CloseSocket
+#endif
+
```

Comment:
moce to compat.h ?

Reply:
Moved here: https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa